bluesky/plans.py

```diff
@@ -90,36 +96,67 @@ def list_scan(detectors, motor, steps, *, per_step=None, md=None):
     --------
     :func:`bluesky.plans.rel_list_scan`
     """
+    if len(args) % 2 != 0:
+            raise ValueError("The list of arguments must contain a list of"
+                             "points for each defined motor")
+
+    # check that all lists are the same length
+    lengths = {}
+    motors = []
+    pos_lists = []
+    for motor, pos_list in partition(2,args):
+        lengths[motor.name] = len(pos_list)
+        motors.append(motor)
+        pos_lists.append(pos_list)
+    lengths = {motor.name: len(pos_list) for motor, pos_list
```

Comment:
I guess this statement is redundant to the loop above.

Reply:
yes, thanks for highlighting. I have removed it